micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/metrics/CamundaMetrics.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package info.novatec.micronaut.camunda.bpm.feature.metrics;
+
+import io.micronaut.context.ApplicationContext;
+import org.camunda.bpm.engine.ManagementService;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.management.MetricsQuery;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+abstract public class CamundaMetrics {
+
+    protected Duration step = Duration.ofMinutes(1);
+
+    protected Instant lastUpdate = Instant.now();
+
+    protected ManagementService managementService = null;
+
+    // Using application context here because injection of managementService or processEngine leads to infinity loop during initialization
+    // metrics are init very early even before datasource is completly setup
+    // but processengine needs datasource
+    protected final ApplicationContext applicationContext;
+
+    protected final Map<String, MetricsQuery> metricsQueryMap = new HashMap<>();
+
+    protected CamundaMetrics(ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    abstract protected void update();
+
+    protected void doUpdateIfNecessary() {
+        if (shouldLoad()) {
+            update();
+        }
+    }
+
+    protected boolean shouldLoad() {
```

Comment:
I don't understand why do you need to check, if the value should be updated? Why in one minute interval? 

A Gauge will be observed after certain amount of time and not continuously.